feathr-example/bureau_bureau-balance.ipynb

```diff
@@ -0,0 +1,1224 @@
+    "    df = df.withColumn(\"TRAN_DATE\", lit(datetime.datetime(2021,1,1,11,34,44).strftime('%Y-%m-%d %X')))\n",
+    "    df_org = df.toPandas()\n",
+    "        \n",
+    "    df_aggCountBureau = aggCountBureau(df_org)\n",
+    "    df_aggAvgInstalments = aggAvgBureau(df_org)\n",
+    "    df_debtCreditRatio = debtCreditRatio(df_org)\n",
+    "    \n",
+    "    dfs = []\n",
+    "\n",
+    "    dfs.append(df_aggCountBureau)\n",
+    "    dfs.append(df_aggAvgInstalments)\n",
+    "    dfs.append(df_debtCreditRatio)\n",
+    "\n",
+    "    df_result = dfs.pop()\n",
+    "    while dfs:\n",
+    "        df_result = df_result.join(dfs.pop(),on='SK_ID_CURR')\n",
+    "    \n",
+    "    # results df would be merge to the original df\n",
+    "    df_result = pd.merge(df_org, df_result, on=\"SK_ID_CURR\", how=\"left\")\n",
+    "    # merging df with same column name would result a columnname with a suffix of `_x` and `_y`.\n",
+    "    # Renaming the column name with suffix `_x` to retain the original column name\n",
+    "    df_result.columns = df_result.columns.str.rstrip(\"_x\")\n",
+    "\n",
+    "    # convert panda to spark dataframe\n",
+    "    spark_session = sql.SparkSession.builder.appName(\"pdf to sdf\").getOrCreate()\n",
+    "    \n",
+    "    return spark_session.createDataFrame(df_result)  \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Defining Features with Feathr:\n",
+    "\n",
+    "### Bureau Dataset\n",
+    "1. parent dataset: bureau.csv \n",
+    "    1. count aggregation features created\n",
+    "    1. average aggregation features created\n",
+    "    1. debt:credit ratio feature created\n",
+    "1. child dataset: bureau_balance.csv\n",
+    "    1. rolling window credit loan status feature will be created and joined to parent dataset\n",
+    "1. combinig/joining both datasets, which will be aggregated in line with primary key (\"SK_ID_CURR) of application_train (target dataframe) with the following features:    \n",
+    "    1. count aggregation features created\n",
+    "    1. average aggregation features created\n",
+    "    1. debt:credit ratio feature created\n",
+    "    1. rolling window credit loan status feature will be created and joined to parent dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# two datasource pointing to same csv, limitation that you could not mix\n",
+    "# pass through and aggregated features. By separating, it must have different datasource (datasource name)\n",
+    "\n",
+    "# source for pass through features\n",
+    "# \"TRAN_DATE\" column created on on the \"datasource_prepocessing\" method.\n",
+    "bureau_source_core = HdfsSource(name=\"bureauSourceCore\",\n",
+    "                          path=\"abfss://feathrhomecreditcafs@feathrhomecreditcasto.dfs.core.windows.net/home_credit_data/bureau.csv\",\n",
```

Comment:
Please remove references to Azure resources.

Reply:
Moved the resource prefix to a variable.